packages/adblocker-extended-selectors/src/eval.ts

```diff
@@ -71,104 +71,247 @@ export function matches(element: Element, selector: AST): boolean {
       }
 
       return text.length >= minLength;
+    } else if (selector.name === 'matches-path') {
+      const { argument } = selector;
+      if (argument === undefined) {
+        return false;
+      }
+
+      const path = globalThis.window.location.pathname;
+
+      let pattern = argument;
+      if (pattern.startsWith('/') && pattern.endsWith('/')) {
+        pattern = pattern.slice(1, -1);
+      }
+
+      const regex = new RegExp(pattern);
+      return regex.test(path);
+    } else if (selector.name === 'matches-attr') {
+      const { argument } = selector;
+      if (argument === undefined) {
+        return false;
+      }
+
+      const indexOfEqual = argument.indexOf('=');
+      let namePattern, valuePattern;
+      if (indexOfEqual === -1) {
+        namePattern = argument;
+      } else {
+        namePattern = argument.slice(0, indexOfEqual);
+        valuePattern = argument.slice(indexOfEqual + 1);
+      }
+
+      let value;
+      if (namePattern.startsWith('/') && namePattern.endsWith('/')) {
+        // matching attribute name by regex
+        const regex = new RegExp(namePattern.slice(1, -1));
+        const attribute = [...element.attributes].find((attr) => regex.test(attr.name));
```

Comment:
I'd recommend preferring to `for of` loop since it's based on iterator.

Reply:
It should not matter here as arguments list is typically small